spec/support/helpers/api_helper.rb

```diff
@@ -135,6 +135,12 @@ def stub_multiple_offenders(offenders, bookings)
       to_return(body: bookings.to_json)
   end
 
+  # Stub an 'empty' response from the Prison API, indicating that the offender does not exist in NOMIS
+  def stub_non_existent_offender(offender_no)
+    stub_request(:get, "#{T3}/prisoners/#{offender_no}").
```

Comment:
is this really what the Prison API does? Does it not 404?

Reply:
@starswan It feels odd, but yes: this is what the Prison API does.

The `/prisoners/{offenderNo}` endpoint returns an array of offenders matching the specified offender number. So in practice, we either return 0 or 1 elements in the array depending on if the offender was found or not.

This is documented in the [Prison API docs](https://api-dev.prison.service.justice.gov.uk/swagger-ui.html), but also I've verified that this is the actual behaviour of the API in production:

```
GET /api/prisoners/123abc (0.087 s)
=> []
```

Reply:
I think this needs a clearer comment - and maybe moving into its own helper method as well just to enphasise the point. Despite the documentation, IMHO this is a violation of the 'principle of least surprise'